### PR TITLE
Dynamically set shards to the number of kube nodes

### PR DESCRIPTION
**Pull request by derekchiang**
...when it's not specified by the user

### CODE REVIEW

**Comment by derekchiang** [comment]
Fix #210

**Comment by derekchiang** [comment]
@jdoliner

**Comment by derekchiang** [comment]
Ideally in the test, I want to check if the number of jobs spawned are indeed equal to the number of nodes, but I'm not sure what a clean way to obtain a kube client in the test is.  Thoughts?

File: src/server/pps/server/api_server.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ import (
 	"k8s.io/kubernetes/pkg/api/unversioned"
 	"k8s.io/kubernetes/pkg/apis/extensions"
 	kube "k8s.io/kubernetes/pkg/client/unversioned"
+	kubeAPI "k8s.io/kubernetes/pkg/api"
```
Comment (jdoliner) [inline src/server/pps/server/api_server.go]: Camel case for packages is a bit weird, could we do `kube_api`?

**Comment by jdoliner** [comment]
You could just do `InspectJob` it'll have the shards which is a good enough check in my opinion.

**Comment by derekchiang** [comment]
@jdoliner comments addressed.  So from `JobInfo` I can check if `shards > 0`, but ideally I want to check that `shards` is actually equal to the number of kube nodes.  Maybe this will do for now though.

**Comment by jdoliner** [comment]
Ahh, I see. You should be able to construct a k8s client using the `New` method here:
k8s.io/kubernetes/pkg/client/unversioned/helper.go.
You'll have to set a few flags to get it to work insecurely.
I don't think it's necessary for merging this PR though.
This looks mergable as is.

**Pull request merged**